src/lumigo_tracer/spans_container.py

```diff
@@ -226,6 +226,12 @@ def end(self, ret_val=None) -> Optional[int]:
                     "The lambda will probably fail due to bad return value. " + suffix,
                 )
         self.function_span.update({"return_value": parsed_ret_val})
+        if _is_span_has_error(self.function_span):
+            self.function_span["envs"] = _get_envs_for_span(has_error=True)
+            if event:
+                self.function_span["event"] = EventDumper.dump_event(
+                    copy.deepcopy(event), has_error=True
```

Comment:
I'm doing here `deepcopy` because this is what we do in the function `create_span`.
The other option is to `deepcopy` at the calling functions, I think it will be worse because this means we'll keep a copy of the event in the memory for the entire execution of lambda although in most cases the event will be used only at the beginning.

Reply:
agree that it should be here.

I'm not sure about its relevance - because the function has already been finished, so what's the point of not changing the event?
On the other hand, some other "wrapper" could run after us, and we might change this value 🤔

Reply:
don't you want to do this logic even if one of the HTTPs has failed?
take it one line below and use `spans_contain_errors`.
Moreover, this function gets fat. Can you extract it to a method?

Reply:
About the relevance of the event - there are other cases as well, like returning the event as `return_value` of the lambda.

About "don't you want to do this logic even if one of the HTTPs has failed?" - I consulted with Efi and we decided to do it only if the lambda fails.

Reply:
ohh of course that only if lambda fails,
But why don't you just pass it to the SpansContainer - and it will call the EventDumper by itself?